meson.build

```diff
@@ -1005,7 +1013,7 @@ libbpf = dependency('libbpf', required : bpf_framework_required, version : '>= 0
 conf.set10('HAVE_LIBBPF', libbpf.found())
 
 if want_bpf_framework == 'false' or not libbpf.found()
-        conf.set10('BPF_FRAMEWORK', 0)
+        conf.set10('BPF_FRAMEWORK', false)
```

Comment:
Looks like the CI didn't catch it because the bpf framework ignores `skip-deps`. I think maybe `skip-deps` should be added to the condition by analogy with all the other dependencies.

Regarding `version_compare('>= 0.55')` as usual I think `meson` should be bumped :-) I'm not sure why upstream projects should cater to slowly moving distributions trying to build the latest systemd with outdated toolchains.

Reply:
> Looks like the CI didn't catch it because the bpf framework ignores `skip-deps`. I think maybe `skip-deps` should be added to the condition by analogy with all the other dependencies.

Added one more commit for that.

> Regarding `version_compare('>= 0.55')` as usual I think `meson` should be bumped :-) I'm not sure why upstream projects should cater to slowly moving distributions trying to build the latest systemd with outdated toolchains.

Yeah, I agree, but let's reconsider that after v251.

Reply:
> Added one more commit for that.

Thanks! I think it should help to set `BPF_FRAMEWORK` to `false`. `HAVE_LIBBPF` can still be true as far as I understand and some code relying on libbpf can still be compiled:
```sh
$ git grep HAVE_LIBB
meson.build:conf.set10('HAVE_LIBBPF', libbpf.found())
src/analyze/analyze-filesystems.c:#if ! HAVE_LIBBPF
src/core/dbus-execute.c:#if HAVE_LIBBPF
src/core/execute.c:#if HAVE_LIBBPF
src/core/execute.c:#if HAVE_LIBBPF
src/core/execute.c:#if HAVE_LIBBPF
src/core/execute.c:#if HAVE_LIBBPF
src/core/main.c:#if HAVE_LIBBPF
src/core/manager.c:#if HAVE_LIBBPF
src/shared/bpf-dlopen.c:#if HAVE_LIBBPF
src/shared/bpf-dlopen.h:#if HAVE_LIBBPF
src/shared/meson.build:if conf.get('HAVE_LIBBPF') == 1
src/test/test-dlopen-so.c:#if HAVE_LIBBPF
```

I think technically it has nothing to do with the warning and the CI so it can be fixed later. Thanks!